src/load/migrate-database.lisp

```diff
@@ -472,8 +492,21 @@
 
                      (when (and create-indexes
                                 (zerop (gethash table writers-count)))
-                       (log-message :notice "DONE copying ~a"
-                                    (format-table-name table))
+
+                       (let* ((stats   pgloader.monitor::*sections*)
+                              (section (get-state-section stats :data))
+                              (table-stats (pgstate-get-label section table))
+                              (pprint-secs
+                               (pgloader.state::format-interval seconds nil)))
+                         ;; in CCL we have access to the *sections* dynamic
+                         ;; binding from another thread, in SBCL we access
+                         ;; an empty copy.
```

Comment:
Do we need a singleton object accessible from all threads in order to access some monitor data? If that's the case, we can use https://github.com/lmj/global-vars that is well-suited for that use case.

Reply:
That's good thinking, and I think that's relevant for another PR. The PR introducing the statistics singleton should also take care of the following bits of refactoring of the code:

  - the singleton is named “state” as I initially though I would update the global state of the  migration to be able to know about the progress being made while it's being made,
  - when implementing the feature, it quickly enough turned into statistics (count of rows read, imported, errors, timings, etc) rather “state”
  - the API to access and maintain the state objects (stats really) per table etc is rather lacking, so we should clean that up and offer easy access to needed information from threads

At the moment only the monitor thread is creating and updating the data, we certainly want to keep it that way. That allows no-locks (and no dead locks) code, with using a queue to send events and centralise statistics from.